### PR TITLE
Localconfig

### PR DESCRIPTION
**Pull request by gidden**
This PR does a lot of stuff:

- remove dynamic `default_paths.py` generation
- remove `config` directory
- allow user-defined configuration files and directories via `ixmp-config` cli
  - now with tests!
- adds a logger
- clean up some `core.py` code
- version bump to `0.1.1`

### CODE REVIEW

File: ixmp/core.py [hunk 1]
```diff
@@ -90,26 +81,25 @@ def __init__(self, dbprops=None, dbtype=None, jvmargs=None):
...
+                logger().info("launching ixmp.Platform using config file at '{}'"
+                              .format(dbprops))
                 self._jobj = java.ixmp.Platform("Python", dbprops)
             # if dbtype is specified, launch Platform with local database
-            else:
-                dbprops = dbprops or os.path.join(local_path,
-                                                  'localdb', 'default')
-                print("launching ixmp.Platform with local {} database at '{}'"
-                      .format(dbtype, dbprops))
+            elif dbtype == 'HSQLDB':
```
Comment (gidden) [inline ixmp/core.py]: So I added this explicit check for HSQLDB. We can take it out, but I think it's worth it (esp if we get other backends). I do *not* understand how this plays with our tests, though... see comment below.

File: tests/testing_utils.py [hunk 2]
```diff
@@ -45,6 +48,60 @@ def test_mp():
...
+    # configure
+    cmd = 'ixmp-config --db_config_path {}'.format(dirname)
+    subprocess.check_call(cmd.split())
+
+    # start jvm
+    ixmp.start_jvm()
+
+    # launch Platform and connect to testdb (reconnect if closed)
+    try:
+        mp = ixmp.Platform(basename)
```
[the remark this answers is not in the document]
Comment (gidden) [reply tests/testing_utils.py]: Same with the command below..

**Comment by gidden** [comment]
This should be good to go @danielhuppmann . We should probably update our documentation to include the CLIs, but I suggest we do a doc push after all features are in..

**Comment by gidden** [comment]
Ok, finally actually good to go @danielhuppmann. Let me know what you think on the issue I raised above, though!

**Comment by danielhuppmann** [comment]
thanks @gidden ! will review in detail tomorrow, but about the problem with `dbtype='HSQLDB'`: if I understand this correctly, you tried to provide a config file and the dbtype - that's overdoing it.

You can either
 - provide as `dbprops` the path to a config file which contains the path and the db type
 - provide the path to a local hsqldb file as `dbprops` and set `dbtype='HSQLDB'`

**Comment by gidden** [comment]
Ok @danielhuppmann, I now explicitly check for this to prevent further errors (and tests added!)

**Comment by gidden** [comment]
To note... on your second point, you can not do this. this is what I tried and it raised an error further in the java pipeline.

**Comment by gidden** [comment]
Ok, looks like I was wrong. I was previously giving it a path to a dbconfig file and the dbtype. This is still funny how it works...

**Comment by danielhuppmann** [comment]
lgtm, all checks pass when installing on Windows!

**Pull request merged**